ibis/backends/__init__.py

```diff
@@ -926,8 +926,8 @@ def compile(
         """Compile an expression."""
         return self.compiler.to_sql(expr, params=params)
 
-    def _to_sql(self, expr: ir.Expr, **kwargs) -> str:
```

Comment:
There are still 2 occurencies where `_to_sql` is used:

```
❯ rg "\._to_sql\("
ibis/backends/duckdb/__init__.py
1374:        query = self._to_sql(expr, params=params)
1410:        query = self._to_sql(expr, params=params)
```

These should be removed in favor of `self.compile()`

Reply:
Also `_to_sql()` is defined at 3 other places:

```
❯ rg "def _to_sql\("
ibis/tests/expr/mocks.py
56:    def _to_sql(self, expr, **kwargs):

ibis/backends/sql/__init__.py
123:    def _to_sql(self, expr: ir.Expr, **kwargs) -> str:

ibis/backends/flink/__init__.py
336:    def _to_sql(self, expr: ir.Expr, **kwargs: Any) -> str:
```

Reply:
I think removing the `transpile()` surfaced an issue in the compiler:

BEFORE TRANSPILE

```py
Select(
  expressions=[
    Alias(
      this=In(
        this=Column(
          this=Identifier(this=x, quoted=True),
          table=Identifier(this=t0, quoted=True)),
        expressions=[  # here is the difference
          Select(
            expressions=[
              Column(
                this=Identifier(this=x, quoted=True),
                table=Identifier(this=t0, quoted=True))],
            from=From(
              this=Table(
                this=Identifier(this=t, quoted=True),
                alias=Identifier(this=t0, quoted=True))),
            where=Where(
              this=GT(
                this=Column(
                  this=Identifier(this=x, quoted=True),
                  table=Identifier(this=t0, quoted=True)),
                expression=Literal(this=2, is_string=False))))]),
      alias=Identifier(this=InSubquery(x), quoted=True))],
  from=From(
    this=Table(
      this=Identifier(this=t, quoted=True),
      alias=Identifier(this=t0, quoted=True))))
```

AFTER TRANSPILE: 

```py
Select(
  expressions=[
    Alias(
      this=In(
        this=Column(
          this=Identifier(this=x, quoted=True),
          table=Identifier(this=t0, quoted=True)),
        query=Select(  # here is the difference
          expressions=[
            Column(
              this=Identifier(this=x, quoted=True),
              table=Identifier(this=t0, quoted=True))],
          from=From(
            this=Table(
              this=Identifier(this=t, quoted=True),
              alias=TableAlias(
                this=Identifier(this=t0, quoted=True)))),
          where=Where(
            this=GT(
              this=Column(
                this=Identifier(this=x, quoted=True),
                table=Identifier(this=t0, quoted=True)),
              expression=Literal(this=2, is_string=False))))),
      alias=Identifier(this=InSubquery(x), quoted=True))],
  from=From(
    this=Table(
      this=Identifier(this=t, quoted=True),
      alias=TableAlias(
        this=Identifier(this=t0, quoted=True)))))
```

Reply:
Here is the fix:

```diff
diff --git a/ibis/backends/sql/compiler.py b/ibis/backends/sql/compiler.py
index e32ff028c..a929e2afc 100644
--- a/ibis/backends/sql/compiler.py
+++ b/ibis/backends/sql/compiler.py
@@ -925,7 +925,7 @@ class SQLGlotCompiler(abc.ABC):
         return self.f.exists(select)

     def visit_InSubquery(self, op, *, rel, needle):
-        return needle.isin(rel.this)
+        return needle.isin(query=rel.this)

     def visit_Array(self, op, *, exprs):
         return self.f.array(*exprs)
```

Reply:
Thanks, should cut down on the snapshot changes too.